system/jlib/jfile.cpp

```diff
@@ -3755,6 +3763,13 @@ bool CFile::getInfo(bool &isdir,offset_t &size,CDateTime &modtime)
 
 #endif
 
+IDirectoryIterator *CFile::directoryFilesSorted(SortDirectoryMode mode, bool rev, const char *mask, bool sub, bool includedirs)
+{
+    IDirectoryIterator *baseIter = directoryFiles(mask, sub, includedirs);
+    if (SD_nosort == mode)
+        return baseIter;
+    return new CSortedDirectoryIterator(*baseIter, mode, rev, includedirs);
+}
```

Comment:
Am wondering since this is a universal implementation, if it wouldn't be better to implement this in a an abstract CFileBase class.
i.e. the git, archive, sockfile impls. can inherit from that rather than re-implement it.

Reply:
Ok, I will implement this in new CFileBase. Since the new directoryFilesSorted() will be removed from the git, archive, etc., I think it would be clear if I rebase the changes into 1 commit for you to review.